### ISSUE TITLE
Play a mp3 file through the mic

### ISSUE THREAD

**Issue by PabloHorno** [naudio/NAudio#370]
I'm trying to play a MP3 file like if it was recorded through the mic, but I'm not able to do so...
I think it could be done by capturing the mic sound and then mix the recorded wave of the mic with the mp3. But I don't know how to play that mixed wave like if it were recored through the mic.

- [x] `Thank you so much!!` 

( _Right now I'm using Virtual Cable so I can play a file to the cabled output and will be played on the cabled input, but I'm pretty sure Naudo can do it better and I just want to use this amazing library:heart_eyes:_ )

**Comment by Cor4xX** [comment]
Up, need it too

**Comment by markheath** [comment]
Unfortunately Windows does not enable you to pipe audio directly to a soundcard input, so something like Virtual Cable is actually your only option here.

**Comment by Cor4xX** [comment]
Yeah you're right and i'm currently playing with Virtual cable. I can play mp3 on my virtual cable output who's playing the sound on my virtual input.
Do you know if i can play both on virtual cable and headset ? Thanks

**Comment by markheath** [comment]
I am currently using AI to help catch up on the backlog of open issues, and am closing this as there is nothing more we can do at this time. Here's a quick summary of the issues...

This isn't something NAudio can do, and it isn't an NAudio limitation — it's a Windows limitation. Windows has no supported API to inject audio into a microphone / capture device. NAudio can render audio to output devices and capture from input devices, but it cannot make played audio appear as if it came from a microphone.

To present audio as a microphone to other applications you need a virtual audio driver that exposes a virtual *capture* endpoint — for example VB-Audio Virtual Cable (which you've already found) or VB-CABLE. You play your MP3 to the virtual cable's output device with NAudio, and any app that selects the virtual cable's input as its microphone will then receive that audio. That is the correct and
intended approach; it's not a workaround NAudio can replace.

For the follow-up question about playing to both the virtual cable and your headset at the same time: open two output devices in NAudio and start playback on both simultaneously (or use a buffered/tee approach feeding both). Some virtual-cable products also have built-in monitoring/routing options that do this for you.

Since the original question has been fully answered and the underlying limitation is in Windows rather than NAudio, we're going to close this issue. If you have a more specific NAudio playback/mixing question (for example mixing the mic capture with the MP3 before sending it to the virtual cable), please open a new issue with details and we'll be glad to help.
